projects/schematics/src/shared/utils/lib-utils.ts

```diff
@@ -603,29 +632,73 @@ export function addPackageJsonDependencies(
 
 export function addPackageJsonDependenciesForLibrary<
   OPTIONS extends LibraryOptions
+>(dependencies: Record<string, string>, options: OPTIONS): Rule {
+  return (tree: Tree, context: SchematicContext): Rule => {
+    const packageJson = readPackageJson(tree);
+    const spartacusLibraries = createSpartacusDependencies(dependencies);
+    const thirdPartyLibraries = createDependencies(dependencies);
+    const libraries = spartacusLibraries.concat(thirdPartyLibraries);
+
+    const cliFeatures = spartacusLibraries
+      .map((dependency) => dependency.name)
+      .reduce((previous, current) => {
+        return {
+          ...previous,
+          // just install the Spartacus library, without any sub-features
+          [current]: [],
+        };
+      }, {} as Record<string, string[]>);
+    const featureOptions = createSpartacusFeatureOptionsForLibrary(
+      options,
+      cliFeatures
+    );
+    addSchematicsTasks(featureOptions, context);
+
+    return chain([
+      addPackageJsonDependencies(libraries, packageJson),
+      installPackageJsonDependencies(),
+    ]);
+  };
+}
+
+function installRequiredSpartacusFeatures<
+  OPTIONS extends LibraryOptions
 >(options: {
-  packageJson: any;
-  context: SchematicContext;
-  dependencies: Record<string, string>;
+  dependencyManagement: DependencyManagement;
   options: OPTIONS;
 }): Rule {
-  const spartacusLibraries = createSpartacusDependencies(options.dependencies);
-  const thirdPartyLibraries = createDependencies(options.dependencies);
-  const libraries = spartacusLibraries.concat(thirdPartyLibraries);
+  return (_tree: Tree, context: SchematicContext): Rule => {
+    const dependencyManagement = options.dependencyManagement;
+    if (!dependencyManagement) {
+      return noop();
+    }
 
-  const dependencyRule = addPackageJsonDependencies(
-    libraries,
-    options.packageJson
-  );
+    logFeatureInstallation(dependencyManagement, context);
+    const featureOptions = createSpartacusFeatureOptionsForLibrary(
+      options.options,
+      dependencyManagement.featureDependencies
+    );
+    addSchematicsTasks(featureOptions, context);
 
-  const featureOptions = createSpartacusFeatureOptionsForLibrary(
-    spartacusLibraries.map((dependency) => dependency.name),
-    options.options
-  );
-  addSchematicsTasks(featureOptions, options.context);
+    return installPackageJsonDependencies();
```

Comment:
Do we need to install packages here? Everything should be installed after `addPackageJsonDependenciesForLibrary`

Reply:
Removed